File: kotlin/src/main/kotlin/com/cardinal/sdk/CreateSdk.kt
package com.cardinal.sdk

import com.icure.sdk.IcureSdk
import com.icure.sdk.auth.UsernamePassword
import com.icure.sdk.options.AuthenticationMethod
import com.icure.sdk.storage.impl.FileStorageFacade

private const val ICURE_URL = "https://api.icure.cloud"


suspend fun createSdk(username: String, password: String): IcureSdk = IcureSdk.initialize(
	applicationId = null,
	baseUrl = ICURE_URL,
	authenticationMethod = AuthenticationMethod.UsingCredentials(
		UsernamePassword(username, password)
	),
	baseStorage = FileStorageFacade("./scratch/storage")
)

File: python/src/create_sdk.py
from icure import IcureSdk
from icure.authentication.AuthenticationMethod import UsernamePassword
from icure.storage.StorageFacadeOptions import FileSystemStorage

ICURE_URL = "https://api.icure.cloud"


def create_icure_sdk(username: str, password: str) -> IcureSdk:
    return IcureSdk(
        application_id=None,
        baseurl=ICURE_URL,
        authentication_method=UsernamePassword(username, password),
        storage_facade=FileSystemStorage("./scratch/storage")
    )
